File: backend/app/schemas/skills.py
from pydantic import BaseModel
from typing import Optional

class SkillCreate(BaseModel):
    skill_options_id: str
    category:Optional[str] = ""
    rank: Optional[int] = None

class SkillUpdate(BaseModel):
    skill_options_id: str
    category: Optional[str] = ""
    rank: Optional[int] =  None

class SkillPatch(BaseModel):
    skill_options_id: Optional[str] = None
    category: Optional[str] =  None
    rank: Optional[int] =  None

class SkillResp(BaseModel):
    id: str
    name: str
    default_category: str | None
    skill_options_id: str
    category: str
    rank: int | None
    class Config:
        orm_mode = True

class SkillOptionCreate(BaseModel):
    name: str
    default_category: str | None = ""

class SkillOptionUpdate(BaseModel):
    name:  Optional[str] =  None
    default_category:  Optional[str] =  None

class SkillOptionsResp(BaseModel):
    id: str
    name: str
    default_category: str  | None
    class Config:
        orm_mode = True

File: frontend/src/types/skills.ts
export interface Skill {
  label: string;
  value: string;
  alias?: Array<string>;
}

export type SkillsList = Array<Skill>;

export interface Skill {
  id: string;
  skillOptionsId: string;
  category: string;
  rank: number | null;
  name: string;
  defaultCategory: string | null;
}

export interface SkillCreate {
  skillOptionsId: string;
  category?: string;
  rank?: number | null;
}

export interface SkillUpdate {
  skillOptionsId: string;
  category?: string;
  rank?: number | null;
}

export interface SkillOption {
  id: string;
  name: string;
  defaultCategory: string | null;
}

export interface SkillOptionCreate {
  name: string;
  defaultCategory?: string;
}

// Skill Select UI
export interface SkillSelect {
  label: string;
  value: string;
  id: string;
  category: string;
  rank: number | null;
}

export interface SkillGroup {
  id: "frontend" | "backend" | "database" | "general" | "";
  name: string;
}
